packages/-ember-data/tests/unit/adapters/build-url-mixin/build-url-test.js

```diff
@@ -18,144 +18,143 @@ module('unit/adapters/build-url-mixin/build-url - DS.BuildURLMixin#buildURL', {
+
+  test('buildURL - findAll requestType delegates to urlForFindAll', function(assert) {
+    assert.expect(3);
+    let originalMethod = adapter.urlForFindAll;
+    let snapshotStub = { snapshot: true };
+    adapter.urlForFindAll = function(type, snapshot) {
+      assert.equal(type, 'super-user');
+      assert.equal(snapshot, snapshotStub);
+      return originalMethod.apply(this, arguments);
+    };
+    assert.equal(adapter.buildURL('super-user', null, snapshotStub, 'findAll'), '/superUsers');
+  });
+
+  test('buildURL - query requestType delegates to urlForQuery', function(assert) {
+    assert.expect(3);
+    let originalMethod = adapter.urlForQuery;
+    let queryStub = { limit: 10 };
+    adapter.urlForQuery = function(query, type) {
+      assert.equal(query, queryStub);
+      assert.equal(type, 'super-user');
+      return originalMethod.apply(this, arguments);
+    };
+    assert.equal(adapter.buildURL('super-user', null, null, 'query', queryStub), '/superUsers');
+  });
+
+  test('buildURL - queryRecord requestType delegates to urlForQueryRecord', function(assert) {
+    assert.expect(3);
+    let originalMethod = adapter.urlForQueryRecord;
+    let queryStub = { companyId: 10 };
+    adapter.urlForQueryRecord = function(query, type) {
+      assert.equal(query, queryStub);
+      assert.equal(type, 'super-user');
+      return originalMethod.apply(this, arguments);
+    };
+    assert.equal(
+      adapter.buildURL('super-user', null, null, 'queryRecord', queryStub),
+      '/superUsers'
+    );
+  });
+
+  test('buildURL - findMany requestType delegates to urlForFindMany', function(assert) {
+    assert.expect(3);
+    let originalMethod = adapter.urlForFindMany;
+    let idsStub = [1, 2, 3];
+    adapter.urlForFindMany = function(ids, type) {
+      assert.equal(ids, idsStub);
+      assert.equal(type, 'super-user');
+      return originalMethod.apply(this, arguments);
+    };
+    assert.equal(adapter.buildURL('super-user', idsStub, null, 'findMany'), '/superUsers');
+  });
+
+  test('buildURL - findHasMany requestType delegates to urlForFindHasMany', function(assert) {
+    assert.expect(4);
+    let originalMethod = adapter.urlForFindHasMany;
+    let snapshotStub = { snapshot: true };
+    adapter.urlForFindHasMany = function(id, type, snapshot) {
+      assert.equal(id, 1);
+      assert.equal(type, 'super-user');
+      assert.equal(snapshot, snapshotStub);
+      return originalMethod.apply(this, arguments);
+    };
+    assert.equal(adapter.buildURL('super-user', 1, snapshotStub, 'findHasMany'), '/superUsers/1');
+  });
+
+  test('buildURL - findBelongsTo requestType delegates to urlForFindBelongsTo', function(assert) {
+    assert.expect(4);
+    let originalMethod = adapter.urlForFindBelongsTo;
+    let snapshotStub = { snapshot: true };
+    adapter.urlForFindBelongsTo = function(id, type, snapshot) {
+      assert.equal(id, 1);
+      assert.equal(type, 'super-user');
+      assert.equal(snapshot, snapshotStub);
```

Comment:
@pete-the-pete want to look into whether there exists a lint rule we can add for `qunit` asserts to enforce that the asserts have labels?

eg. not `assert.ok(bool)` but `assert.ok(bool, 'some-label')` and not `assert.equal('a', 'b')` but `assert.equal('a', 'b', 'some-description')`

Reply:
Opened an issue to track: https://github.com/emberjs/data/issues/6090